tests/mbt/model/ccv_pss_model.qnt

```diff
@@ -0,0 +1,113 @@
+module ccv_pss_model {
+    import ccv_types.* from "./ccv"
+    import ccv_model.* from "./ccv_model"
+    import ccv_pss.* from "./ccv_pss"
+    import extraSpells.* from "./libraries/extraSpells"
+
+    action StepOptIn(): bool = {
+        all {
+            runningConsumers.size() > 0,
+            nondet consumer = oneOf(runningConsumers)
+            nondet validator = oneOf(nodes)
+            OptIn_Deterministic(consumer, validator)
+        }
+    }
+
+    action OptIn_Deterministic(consumer: Chain, validator: Node): bool = {
+        val res = OptIn(currentState, consumer, validator)
+        all {
+            currentState' = res.newState,
+            trace' = trace.append(
+                {
+                    ...emptyAction,
+                    kind: "OptIn",
+                    consumerChain: consumer,
+                    validator: validator,
+                    expectedError: res.error
+                }
+            ),
+            params' = params,
+        }
+    }
+
+    action StepOptOut(): bool = {
+        all {
+            runningConsumers.size() > 0,    
+            nondet consumer = oneOf(runningConsumers)
+            nondet validator = oneOf(nodes)
+            OptOut_Deterministic(consumer, validator)
+        }
+    }
+
+    action OptOut_Deterministic(consumer: Chain, validator: Node): bool = {
+        val res = OptOut(currentState, consumer, validator)
+        all {
+            currentState' = res.newState,
+            trace' = trace.append(
+                {
+                    ...emptyAction,
+                    kind: "OptOut",
+                    consumerChain: consumer,
+                    validator: validator,
+                    expectedError: res.error
+                }
+            ),
+            params' = params,
+        }
+    }
+
+    // Different sets of possible values for the topN parameter.
+    val allFullConsumers: Set[int] = Set(100)
+    val allOptIn: Set[int] = Set(0)
+    // only choose a few values for top N here to not make the "edge cases" of 0 and 100 too unlikely
+    val randomTopN: Set[int] = Set(0, 50, 70, 80, 90, 100)
```

Comment:
nit: The value chosen from the set is randomly chosen but not the set itself. It might make sense to just name this something akin to `possibleTopNValues`.

Reply:
Great point